src/App.jsx

```diff
@@ -7,24 +7,40 @@ import RestaurantList from "./components/Main/RestaurantList.jsx"
 import Restaurants from "./data/Restaurant.js";
 import { useState } from "react";
 
-function App() { 
+function App() {
   const [ category, setCategory ] = useState("전체");
   const restaurantList = Restaurants();
 
   const filteredRestaurants = category === "전체" ? restaurantList : restaurantList.filter(
     (restaurant) => restaurant.category === category
   );
 
+  const [ clickRestaurantItem, setClickRestaurantItem ] = useState(null);
+  const [ showDetailModal, setShowDetailModal ] = useState(false);
+  
+  const selectedRestaurant = restaurantList.find(
+      (restaurant) => restaurant.name === clickRestaurantItem
+  )
+
+  const handleOpenModal = (restaurantName) => {
+    setClickRestaurantItem(restaurantName);
+    setShowDetailModal(true);
+  };
+
+  const handleCloseModal = () => {
+    setShowDetailModal(false);
+  }
```

Comment:
```suggestion
  const handleRestaurantItem = (restaurantName) => {
    setClickRestaurantItem(restaurantName);
  };
  
  const handleModal = () =>{
    setShowDetailModal(!showDetailModal);
  }
  
```

이런식으로 하면 toggle로 detailModal이 보여 지는 것을 조절 할 수 있을 것 같네요!
아니면 handleModal에 boolean 값을 전달할 수도 있을 거 같네요!
handleModal(false) 처럼요
근데 저는 지금 코드 대로 하시는게 함수가 명호어떨 때 모달을 열고 닫는지 확실하게 보여서 또 좋은 것 같아요!
각각 장단점이 있는거 같네요 준수님은 어떤 방식이 더 좋아보이시나요? 🙂

Reply:
이 방향이 스텝4에서 조금  변경이 되어서 혜정님께서 작성해주신 것처럼 해도 한번에 통제가 가능해서 좋아보이네요!